packages/ed25519-signature-2020/src/Ed25519Signature2020.ts

```diff
@@ -0,0 +1,325 @@
+    proof.signatureValue = signature;
+    return proof;
+  }
+
+  async createProof({
+    document,
+    purpose,
+    documentLoader,
+    expansionMap,
+    compactProof,
+  }: types.CreateProofOptions) {
+    // build proof (currently known as `signature options` in spec)
+    let proof: any = { '@context': constants.SECURITY_CONTEXT_URL };
+
+    // ensure proof type is set
+    proof.type = this.type;
+
+    // set default `now` date if not given in `proof` or `options`
+    let date = this.date;
+    if (proof.created === undefined && date === undefined) {
+      date = new Date().toISOString();
+    }
+
+    // ensure date is in string format
+    if (date !== undefined && typeof date !== 'string') {
+      date = new Date(date).toISOString();
+    }
+
+    // add API overrides
+    if (date !== undefined) {
+      proof.created = date;
+    }
+    // `verificationMethod` is for newer suites, `creator` for legacy
+    if (this.verificationMethod !== undefined) {
+      proof.verificationMethod = this.verificationMethod;
+    }
+
+    // add any extensions to proof (mostly for legacy support)
+    proof = await this.updateProof({
+      proof,
+    });
+
+    // allow purpose to update the proof; the `proof` is in the
+    // SECURITY_CONTEXT_URL `@context` -- therefore the `purpose` must
+    // ensure any added fields are also represented in that same `@context`
+    proof = await purpose.update(proof, {
+      document,
+      suite: this,
+      documentLoader,
+      expansionMap,
+    });
+
+    // create data to sign
+    const verifyData = await this.createVerifyData({
+      document,
+      proof,
+      documentLoader,
+      expansionMap,
+      compactProof,
+    });
+
+    // sign data
+    proof = await this.sign({
+      verifyData,
+      proof,
+    });
+
+    return proof;
+  }
+
+  async getVerificationMethod({
+    proof,
+    documentLoader,
+  }: types.GetVerificationMethodOptions) {
+    let { verificationMethod } = proof;
+
+    if (typeof verificationMethod === 'object') {
+      verificationMethod = verificationMethod.id;
+    }
+
+    if (!verificationMethod) {
+      throw new Error('No "verificationMethod" found in proof.');
+    }
+
+    // Note: `expansionMap` is intentionally not passed; we can safely drop
+    // properties here and must allow for it
+    const framed = await jsonld.frame(
+      verificationMethod,
+      {
+        '@context': constants.SECURITY_CONTEXT_URL,
+        '@embed': '@always',
+        id: verificationMethod,
+      },
+      { documentLoader, compactToRelative: false }
+    );
+
+    if (!framed) {
+      throw new Error(`Verification method ${verificationMethod} not found.`);
+    }
+
+    // Security issue, is the ever implemented like this?
+    // ensure verification method has not been revoked
+    if (framed.revoked !== undefined) {
+      throw new Error('The verification method has been revoked.');
+    }
```

Comment:
Do we actually use revoked for anything? I ask because it seems like the DID Core spec is going towards just not listing the key. I guess we wanted it to strongly signal that the key didn't just disappear, it was known to be compromised or we're strongly asserting that the key shouldn't be used. Don't know if we have test suite coverage for this?

Reply:
If an implementation does not implement this check... its the worst possible outcome... I think this is a foot gun and suggest we remove it from this suite, and mark it deprecated in security-vocab.

Reply:
i removed this.